src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2940,13 +2940,15 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                                 }
                             }
                         }
+                        fullPath.insert(0, L"\'");
+                        fullPath += L"\'";

```

Comment:
If you don't mind doing it in this PR, the below code is kinda whack and could need an improvement. `.insert(0, L"\"");` is an O(n) operation that is entirely avoidable, because the `fullPath` is being appended to `allPathsString` anyways, so we can instead just append the quotes to `allPathsString` directly. Here's what I mean:

```cpp
const auto isWSL = _interactivity.ManglePathsForWsl();

if (isWSL)
{
    // ...
    // the above WSL specific code block but without your new addition
    // ...
}

const auto quotesNeeded = isWSL || fullPath.find(L' ') != decltype(fullPath)::npos;
const auto quotesChar = isWSL ? L'\'' : L'"';

// Append fullPath and also wrap it in quotes if needed
if (needsQuotes)
{
    allPathsString.push_back(quotesChar);
}
allPathsString.append(fullPath);
if (needsQuotes)
{
    allPathsString.push_back(quotesChar);
}
```

Reply:
Thanks for the feedback! Implemented and tested the changes